Client changes

MohamedShatry opened the pull request:
Changes may also reflect changes on the server branch since this branch is dependent on the server branch

MohamedShatry committed:
Add html rendering

MohamedShatry committed:
Add html rendering

MohamedShatry committed:
Add comments div

MohamedShatry committed:
Add comments div

MohamedShatry committed:
Add default run

MohamedShatry committed:
Add default run

MohamedShatry committed:
Add error handling

MohamedShatry committed:
Add error handling

MohamedShatry committed:
Update class names

MohamedShatry committed:
Update class names

MohamedShatry committed:
Update styling for comments

MohamedShatry committed:
Update styling for comments

MohamedShatry committed:
Update element classes

MohamedShatry committed:
Update element classes

MohamedShatry committed:
Add delete feature

MohamedShatry committed:
Update comments page

MohamedShatry committed:
Add input element

MohamedShatry committed:
Add delete button functionality

addarsh-google commented on portfolio/src/main/webapp/comments-script.js:
```diff
@@ -2,6 +2,79 @@
     .then(res => res.json())
-    .then(res => console.log(res))
+    .then(res => {
+        console.log(res);
```
This is probably not needed anymore.

MohamedShatry replied:
Ooops! My bad. Will fix that. Resolved

addarsh-google commented on portfolio/src/main/webapp/comments-script.js:
```diff
@@ -2,6 +2,79 @@
+    })
+    .then(res => res.json())
+    .then(res => {
+        console.log("Got response");
```
Log statements probably not needed anymore.

MohamedShatry replied:
True. Resolved

addarsh-google commented on portfolio/src/main/webapp/comments-script.js:
```diff
@@ -45,7 +45,7 @@ function createCommentElement(comment){
     commentContainer.appendChild(commentTag);
     commentContainer.appendChild(bottom_div);
-
+    
```
Is this commit as intended?

MohamedShatry replied:
Yep. Everything looks intended to me. I'm sorry if I've missed the error

MohamedShatry replied:
It looks like what I intended it to be. I'm sorry f I'm missing something

addarsh-google commented on portfolio/src/main/webapp/comments-script.js:
```diff
@@ -1,7 +1,121 @@
+}
+
+//This function converts the unix timestamp to date and time format
+function getDateTimeFromTimestamp(unixTimeStamp) {
```
is there a library that can do this?

MohamedShatry replied:
I think momentjs can do that. Let me look into importing it

octopus1000 replied:
> This function converts the unix timestamp to date and time **format**

Can you also comment on what specific format you are looking at?
momentjs can be a bit heavy if we are not playing with dates a lot.

Consider using supported format. date.toISOString() if acceptable.

If certain format must be followed, suggest having a test for this function.

MohamedShatry replied:
No particular date format needs to be followed. I've tried to implement toISOString(), however, it seems to be returning time in GMT. Working on fixing it

MohamedShatry replied:
Figured a solution without converting it to ISOString

addarsh-google commented on portfolio/src/main/webapp/comments-script.js:
```diff
@@ -1,7 +1,121 @@
+
+//Add an event listener to the submit button that send the response as JSON to the server
+//Later, this will allow us to add data from cookies
+
```
nit: space between description and callback.

MohamedShatry replied:
Will fix. Resolved

octopus1000 commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -22,27 +28,72 @@
         String json = gson.toJson(comments);
-
+ 
         // Send the JSON as the response
```
A general comment about "commenting", usually we use a sentence end with period.

//  Send the JSON as response.

MohamedShatry replied:
Did not know this. Will implement

MohamedShatry committed:
Update date rendering

MohamedShatry committed:
Resolve conflicts

MohamedShatry committed:
Update for rebasing

MohamedShatry commented:
This branch has several commits. Most of these might be duplicate because I ran into some trouble with rebasing my client server into this one. 
Some changes implemented on the client side include:
-Adding an input to limit the number of comments to be received. 
-Updated the way I handled dates in javascript
-Added delete button functionality
Updated fetch requests to include parameters

Live version ca be found here : https://mshatry-step-2020.uc.r.appspot.com/

MohamedShatry committed:
Merge branch 'master' into client

MohamedShatry committed:
Update delete server route

MohamedShatry commented:
Merge master branch into client branch. 
Update the client branch to call new delete route

octopus1000 commented on portfolio/src/main/webapp/comments-script.js:
```diff
@@ -1,7 +1,138 @@
+}
+
+
+//Add an event listener to the submit button that send the response as JSON to the server.
```
Do you need to use event listener? Can we just to submit=func() on form element?

MohamedShatry replied:
I can do that. Let me update

MohamedShatry committed:
Add funciton call on click

MohamedShatry committed:
Change event listener to function call

MohamedShatry commented:
Changed the event listener to a function. Also updated delete path method
